src/main/java/uk/gov/ons/census/action/schedule/ActionInstructionBuilder.java

```diff
@@ -0,0 +1,148 @@
+
+@Component
+public class ActionInstructionBuilder {
+  private static final String HOUSEHLD_INITIAL_CONTACT_QUESTIONNIARE_TREATMENT_CODE_PREFIX = "HH_Q";
+  private static final String WALES_TREATMENT_CODE_SUFFIX = "W";
+  private static final int NUM_OF_UAC_IAC_PAIRS_NEEDED_BY_A_WALES_INITIAL_CONTACT_QUESTIONNAIRE = 2;
+  private static final int THE_NUMBER_ONE_IS_NOW_A_CONSTANT_HURRAH_THIS_IS_MUCH_BETTER = 1;
+  private static final String WALES_IN_ENGLISH_QUESTIONNAIRE_TYPE = "02";
+  private static final String WALES_IN_WELSH_QUESTIONNAIRE_TYPE = "03";
+
+  private final UacQidLinkRepository uacQidLinkRepository;
+
+  @Value("${queueconfig.outbound-exchange}")
+  private String outboundExchange;
+
+  public ActionInstructionBuilder(UacQidLinkRepository uacQidLinkRepository) {
+    this.uacQidLinkRepository = uacQidLinkRepository;
+  }
+
+  public ActionInstruction buildActionInstruction(Case caze, ActionRule actionRule) {
+
+    UacQidTuple uacQidTuple = getUacQidLinks(caze);
+
+    ActionEvent actionEvent = new ActionEvent();
+    actionEvent
+        .getEvents()
+        .add("CASE_CREATED : null : SYSTEM : Case created when Initial creation of case");
+
+    ActionAddress actionAddress = new ActionAddress();
+    actionAddress.setLine1(caze.getAddressLine1());
+    actionAddress.setLine2(caze.getAddressLine2());
+    actionAddress.setLine3(caze.getAddressLine3());
+    actionAddress.setTownName(caze.getTownName());
+    actionAddress.setPostcode(caze.getPostcode());
+    actionAddress.setOrganisationName(caze.getOrganisationName());
+    ActionRequest actionRequest = new ActionRequest();
+    actionRequest.setActionId(UUID.randomUUID().toString());
+    actionRequest.setResponseRequired(false);
+    actionRequest.setActionPlan(actionRule.getActionPlan().getId().toString());
+
+    actionRequest.setActionType(actionRule.getActionType().toString());
+
+    actionRequest.setAddress(actionAddress);
+    actionRequest.setLegalBasis("Statistics of Trade Act 1947");
+    actionRequest.setCaseGroupStatus("NOTSTARTED");
+    actionRequest.setCaseId(caze.getCaseId().toString());
+
+    actionRequest.setPriority(Priority.MEDIUM);
+    actionRequest.setCaseRef(Long.toString(caze.getCaseRef()));
+    actionRequest.setIac(uacQidTuple.getUacQidLink().getUac());
+    actionRequest.setQid(uacQidTuple.getUacQidLink().getQid());
+
+    if (uacQidTuple.getUacQidLinkWales().isPresent()) {
+      actionRequest.setIacWales(uacQidTuple.getUacQidLinkWales().get().getUac());
+      actionRequest.setQidWales(uacQidTuple.getUacQidLinkWales().get().getQid());
+    }
+
+    actionRequest.setEvents(actionEvent);
+    actionRequest.setExerciseRef("201904");
+    actionRequest.setUserDescription("Census-FNSM580JQE3M4");
+    actionRequest.setSurveyName("Census-FNSM580JQE3M4");
+    actionRequest.setSurveyRef("Census-FNSM580JQE3M4");
+    actionRequest.setReturnByDate("27/04");
+    actionRequest.setSampleUnitRef("DDR190314000000516472");
```

Comment:
This probably shouldn't be hard coded any more?

Reply:
we don't have a sample unit ref